src/main/java/datastructure/SegmentDB.java

```diff
@@ -0,0 +1,133 @@
+     * The following tables are created:
+     * segment - PRIMARY KEY:    node_id (INT)
+     *           OTHER:          segment (CLOB, NOT NULL)
+     */
+    private void dbInit() {
+        try {
+            Class.forName("org.h2.Driver");
+            Connection con;
+            con = DriverManager.getConnection(dbLoc);
+            con.prepareStatement("DROP TABLE IF EXISTS segment").execute();
+            con.prepareStatement("CREATE TABLE segment(node_id INT PRIMARY KEY, "
+                    + "segment CLOB NOT NULL)").execute();
+            con.close();
+        } catch (ClassNotFoundException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Inserts a node and edges into the database
+     * based on the provided parameters.
+     *
+     * @param id             the id of the node.
+     * @param segment        the segment of the node.
+     */
+    public void addSegment(final int id,
+                        final String segment) {
+        try {
+            String iSegment =
+                    "INSERT INTO SEGMENT \n"
+                            + "VALUES (?, ?)";
+
+            Class.forName("org.h2.Driver");
+            Connection con = DriverManager.getConnection(dbLoc);
+
+            PreparedStatement insertNode = con.prepareStatement(iSegment);
+            insertNode.setInt(1, id);
+            insertNode.setClob(2, new SerialClob(segment.toCharArray()));
+            insertNode.execute();
+
+            con.close();
+        } catch (ClassNotFoundException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Gets the segment of the provided node
+     * from the database.
+     *
+     * @param id The id of the node.
+     * @return The segment corresponding to the id.
+     */
+    public String getSegment(final int id) {
+        String res = "";
+
+        try {
+            Class.forName("org.h2.Driver");
+            Connection con = DriverManager.getConnection(dbLoc);
+            PreparedStatement stmt = con.prepareStatement(
+                    "SELECT segment FROM SEGMENT WHERE node_id = ?"
+            );
+            stmt.setInt(1, id);
+            ResultSet rs = stmt.executeQuery();
+            rs.first();
+            Clob tempClob = rs.getClob(1);
+            res = tempClob.getSubString(1, (int) tempClob.length());
+            con.close();
+        } catch (ClassNotFoundException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        }
```

Comment:
The exceptions are handled by separate catch blocks because that allows us to perform different actions for specific exceptions. Although this is not currently the case, this allows us to easily do so later.